client/allocrunner/taskrunner/template/template_test.go

```diff
@@ -806,6 +806,69 @@ OUTER:
 	}
 }
 
+// Tests an edge case where a task has multiple templates and the client is restarted.
+// In this case, vault may re-render and overwrite some fields in the first render event
+// but we still want to make sure it causes a restart.
+// We cannot control the order in which these templates are rendered, so this test will
+// exhibit flakiness if this edge case is not properly handled.
```

Comment:
This makes me want a test rig where we can mock CT and step thru its operations deterministically... but that's a ton of work, probably better fixed by pulling the runner in-tree, and _way_ out of scope for this PR. :grin: